### DISCUSSION TITLE
ImportError: cannot import name 'trame' from 'trame.widgets'

### DISCUSSION THREAD

**Discussion by jarrah42** [Kitware/trame#366] [Q&A]
I'm trying to run `python ./04_application/solution.py --port 1234` from the tutorial and get the above error. I've tried python 3.9 and 3.11 with same results.

**Comment by jourdain** [comment]
The tutorial was built during trame v2, where all the widgets were described as dependencies. 

But since the release of trame v3, we have removed those dependencies from the default package to give more control to the user and not force an ever-growing set of widgets onto your project, especially if you don't plan to use them.

In other words, an app now needs to describe its dependencies.  So, for that specific script/app, you will need to install the following

```
pip install trame trame-vtk trame-vuetify trame-components
```

You can find the available set of libraries/widgets [here](https://kitware.github.io/trame/guide/intro/widgets.html).

**Reply by jarrah42** [reply]
Ok thanks, that did it. It would probably be a good idea to update your tutorial so it works with trame v3.

**Reply by jourdain** [reply]
Indeed. Normally we updated it with the proper instruction here: https://kitware.github.io/trame/guide/tutorial/setup.html#setup-environment

But later on we are missing the `pip install trame-components` in the Application section (Just noticed it)

**Reply by jourdain** [reply]
That last part has been fixed in [here](https://kitware.github.io/trame/guide/tutorial/application.html#imports)
